internal/services/connection/datasource_connections.go

```diff
@@ -125,8 +125,7 @@ func (d *ConnectionsDataSource) Read(ctx context.Context, req datasource.ReadReq
 
 	connections, err := d.ConnectionsClient.GetConnections(ctx, state.EnvironmentId.ValueString())
 	if err != nil {
-		resp.Diagnostics.AddError(fmt.Sprintf("Client error when reading %s", d.FullTypeName()), err.Error())
-		return
+		resp.Diagnostics.AddError(fmt.Sprintf("Client error when reading %s", d.FullTypeName()), fmt.Errorf("underlying error: %w", err).Error())
```

Comment:
1. I don't think adding "underlying error:" adds value.  Also you removed the `return` statement.  was that intentional?

Reply:
reverted back. The return can be there. There is no point going with the logic when error  was returned but even with not return statement, there will not no connections so the logic will just finish without going over any connection.